syncer/syncer.go

```diff
@@ -132,8 +132,8 @@ type Syncer struct {
 	lastATXsSynced    atomic.Value
 
 	// awaitATXSyncedCh is the list of subscribers' channels to notify when this node enters ATX synced state
-	awaitATXSyncedCh []chan struct{}
-	muATXSyncedCh    sync.Mutex // protects the slice above from concurrent access
+	awaitATXSyncedCh chan struct{}
+	muATXSyncedCh    sync.RWMutex // protects the channel above from concurrent access
```

Comment:
are you observing race condition in this code?

first of all we don't need a RWMutex here. i prefer Mutex when there is not race condition. it's faster and and simpler to use.
secondly, if you are using single channel, why not initialize it the constructor and there is no need to protect it.

Reply:
The RWMutex allows for a more fine-grained access restriction to the field that holds the channel.

Since RegisterATXSynced only reads the field it can use a `RLock()` allowing multiple concurrent calls without mutual exclusion. Only when `setAtxSynced()` is called all access to the field needs to be blocked until the lock is freed again.

Reply:
whether we agreed on whether RWMutex is necessary is beyond the point.
we don't need a lock here. if the channel is created and never changed within syncer, why need lock to protect it?

Reply:
It is changed; `setATXsynced` assigns the field a new value. Without a mutex I'm getting warnings from the race detector.

I just changed the Mutex to an RWMutex that does the same thing but allows multiple concurrent read accesses?

The field needs to be protected by a Mutex and RWMutex just allows for a more fine grained access control